src/core/emsdevice.cpp

```diff
@@ -1699,14 +1699,33 @@ std::string EMSdevice::get_metrics_prometheus(const int8_t tag) {
     std::string                           result;
     std::unordered_map<std::string, bool> seen_metrics;
 
+    // Helper function to check if a device value type is supported for Prometheus metrics
+    auto is_supported_type = [](uint8_t type) -> bool {
+        return type == DeviceValueType::BOOL || type == DeviceValueType::UINT8 || type == DeviceValueType::INT8
+            || type == DeviceValueType::UINT16 || type == DeviceValueType::INT16 || type == DeviceValueType::UINT24
+            || type == DeviceValueType::UINT32 || type == DeviceValueType::TIME || type == DeviceValueType::ENUM;
+    };
+
+    // Dynamically reserve memory for the result
+    size_t entity_count = 0;
+    for (const auto & dv : devicevalues_) {
+        if (tag >= 0 && tag != dv.tag) {
+            continue;
+        }
+        // only count supported types
+        if (is_supported_type(dv.type)) {
```

Comment:
Better `if (dv.hasValue() && is_supported_type(dv.type)) {`
some boiler types have a lot of entities without values. and you generate the metrics also only for entities with values.